wdl-lint/src/rules/disallowed_declaration_name.rs

```diff
@@ -0,0 +1,345 @@
+        "Disallows declaration names that end with their type name"
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Declaration names should not include their type as a suffix. This makes the code more \
+         verbose and often redundant. For example, use 'counter' instead of 'counterInt' or \
+         'is_active' instead of 'isActiveBoolean'."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Style, Tag::Clarity])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [SyntaxKind]> {
+        Some(&[
+            SyntaxKind::VersionStatementNode,
+            SyntaxKind::InputSectionNode,
+            SyntaxKind::OutputSectionNode,
+            SyntaxKind::BoundDeclNode,
+            SyntaxKind::UnboundDeclNode,
+        ])
+    }
+}
+
+impl Visitor for DisallowedDeclarationNameRule {
+    type State = Diagnostics;
+
+    fn document(
+        &mut self,
+        _: &mut Self::State,
+        reason: VisitReason,
+        _: &Document,
+        _: SupportedVersion,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // Reset the visitor upon document entry
+        *self = Default::default();
+    }
+
+    fn bound_decl(&mut self, state: &mut Self::State, reason: VisitReason, decl: &BoundDecl) {
+        if reason == VisitReason::Enter {
+            check_decl_name(state, &Decl::Bound(decl.clone()), &self.exceptable_nodes());
+        }
+    }
+
+    fn unbound_decl(&mut self, state: &mut Self::State, reason: VisitReason, decl: &UnboundDecl) {
+        if reason == VisitReason::Enter {
+            check_decl_name(
+                state,
+                &Decl::Unbound(decl.clone()),
+                &self.exceptable_nodes(),
+            );
+        }
+    }
+}
+
+/// Check declaration name for type suffixes
+fn check_decl_name(
+    state: &mut Diagnostics,
+    decl: &Decl,
+    exceptable_nodes: &Option<&'static [SyntaxKind]>,
+) {
+    let name = decl.name();
+    let name_str = name.as_str();
+
+    // Get the declaration type
+    let ty = decl.ty();
+
+    // Extract type names to check based on the type
+    let mut type_names = HashSet::new();
+
+    // Handle different type variants
+    match ty {
+        Type::Ref(_) => return, // Skip type reference types (user-defined structs)
+
+        Type::Primitive(primitive_type) => {
+            match primitive_type.kind() {
+                // Skip File and String types as they cause too many false positives
+                PrimitiveTypeKind::File | PrimitiveTypeKind::String => return,
+
+                PrimitiveTypeKind::Boolean => {
+                    // Add the primitive type name
+                    type_names.insert(primitive_type.to_string());
+                    // Also check for "Bool"
+                    type_names.insert("Bool".to_string());
+                }
+
+                PrimitiveTypeKind::Integer => {
+                    type_names.insert(primitive_type.to_string());
```

Comment:
This should check both `integer` and `int`. And this confused me for a moment, as I forgot that `Integer` is shorted to `Int` in WDL, so can you add a comment stating that?

Reply:
This is what confused me as well, but you've actually just added `int` twice, and `integer` still isn't being checked. `primitive_type.to_string()` returns `"Int"`, not `"Integer"` like what might be expected.